### PR TITLE
DCOS-14010: Fix MesosLogView -> MesosLogContainer

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ Depends on this PR: https://github.com/dcos/dcos-ui/pull/1954~~ (merged)

---

This PR fixes issues in the MesosLogContainer (former MesosLogView).
- The view would not behave properly and call the right functions when it needed to: I changed the MesosLogStore to go about logs the same way as SystemLogStore does it, and made it also use `LogView` (which has been tested) instead of having its own broken implementation.
- Fixed the search for the logs, which was not working properly before.

Please set this in your config dev:
```js
mesos: {
  'logging-strategy': 'logrotate'
},
```

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [x] If this is a regression, did you write a test to catch this in the future? However, we need more! This is just to cover the basics!

For reviewers: View diff between PRs [here](https://github.com/dcos/dcos-ui/compare/mlunoe/feature/DCOS-14010-add-container-to-pick-logging-component-and-api...mlunoe/feature/DCOS-14010-fix-mesos-log-view?expand=1)

### CODE REVIEW

**Comment by orlandohohmeier** [comment]
@mlunoe any chance you can separate these fixes to make it easier to review and land them?

**Comment by mlunoe** [comment]
They are separated! This PR depends on #1954!

**Comment by mlunoe** [comment]
Hmmm... the diff seems big, but that's because Github doesn't recognise the rename of `MesosLogView -> `MesosLogContainer` :expressionless:
Ok, I will try and split out the small fixes

**Comment by mlunoe** [comment]
Moved tab highlight fix here: https://github.com/dcos/dcos-ui/pull/1975
Moved go to bottom fix here: https://github.com/dcos/dcos-ui/pull/1976

**Comment by mlunoe** [comment]
The search log part is sort of integrated. It just wasn't passing down needed props to the `LogView`: `highlightText`, `watching` and `onCountChange`.

**Comment by jfurrow** [comment]
I'm experiencing a couple of bugs with the `logrotate` strategy enabled.

1. The UI reports the stdout log isn't available initially, but then it loads.
2. When it loads, the spinner at the top of the screen is visible. If this is the top of the log, I would expect the spinner not to appear. If it's not the top of the log, then I would expect to be able to scroll upwards to reveal the previous log.

Here's a screen recording: https://cl.ly/1u0H2f380G0y

**Comment by MatApple** [comment]
@jfurrow I'm having a hard time reproducing the issues you mention above in https://github.com/dcos/dcos-ui/pull/1974#issuecomment-285187739. Other than that, this PR looks like its functioning as expected 👍 . The logs aren't pushed as often as I'd expect given the while loop I have running but it appears to not be a UI problem from what I can tell. @nLight / @orlandohohmeier it would be great if you guys could please check this further. I give it my label but I strongly encourage somebody else to test this in depth please.

**Comment by orlandohohmeier** [comment]
Please squash 1bc8cbd24b78785afd525312638324c939985eb9 and 3e8958b5c55d171ba49a6e732cc580ba2f44af12 as the changes belong together.

File: plugins/services/src/js/stores/MesosLogStore.js [hunk 1]
```diff
@@ -97,50 +104,57 @@ class MesosLogStore extends GetSetBaseStore {
...
+        slaveID,
+        path,
+        logBuffer.getEnd(),
+        MAX_FILE_SIZE
+      );
+    }
   }
 
-  stopTailing(path) {
+  stopTailing(path, shouldClearData = false) {
```
Comment (orlandohohmeier) [inline plugins/services/src/js/stores/MesosLogStore.js]: Please rename `shouldClearData` to `clearData`  to follow the general naming schema #nit

File: plugins/services/src/js/stores/__tests__/MesosLogStore-test.js [hunk 2]
```diff
@@ -24,23 +27,24 @@ describe('MesosLogStore', function () {
...
     });
 
   });
 
   describe('#stopTailing', function () {
 
-    it('should return an instance of LogBuffer', function () {
-      MesosLogStore.stopTailing('/bar');
-      var logBuffer = MesosLogStore.get('/bar');
+    it('shouldn\'t return an instance of LogBuffer', function () {
```
Comment (orlandohohmeier) [inline plugins/services/src/js/stores/__tests__/MesosLogStore-test.js]: Please add a test case for `clearData` as we need should verify that it's properly working to not risk memory leaks.

File: plugins/services/src/js/components/__tests__/MesosLogView-test.js [hunk 3]
```diff
@@ -1,21 +1,24 @@
 jest.dontMock('../Highlight');
-jest.dontMock('../MesosLogView');
+jest.dontMock('../MesosLogContainer');
```
Comment (nLight) [inline plugins/services/src/js/components/__tests__/MesosLogView-test.js]: shall we rename the file as well?

**Comment by nLight** [comment]
🚨 🚨 🚨 🚨  I gave it second label, but please make sure Orlando's comments are addressed.

File: plugins/services/src/js/components/__tests__/MesosLogView-test.js (hunk 3 above)
Comment (orlandohohmeier) [inline plugins/services/src/js/components/__tests__/MesosLogView-test.js]: ✔️  Done

**Comment by orlandohohmeier** [comment]
run integration tests

**Comment by orlandohohmeier** [comment]
@mlunoe can you please help rebasing this one as it seems to clash with changes made in https://github.com/dcos/dcos-ui/pull/1976

File: plugins/services/src/js/components/__tests__/MesosLogView-test.js (hunk 3 above)
Comment (mlunoe) [inline plugins/services/src/js/components/__tests__/MesosLogView-test.js]: Good catch!

**Comment by jfurrow** [comment]
I can't reproduce the issues I came across yesterday, but the "Go to bottom" button still isn't hiding for me. Otherwise this seems to be working perfectly! Here's a screen recording: https://cl.ly/1K0y3M2l291T/Screen%20Recording%202017-03-09%20at%2010.31%20AM.mov

**Comment by jfurrow** [comment]
I'm merging this!

**Pull request merged**